internal/infrastructure/kubernetes/infra_resource.go

```diff
@@ -167,6 +168,42 @@ func (i *Infra) createOrUpdateDaemonSet(ctx context.Context, r ResourceRender) (
 	return i.Client.ServerSideApply(ctx, daemonSet)
 }
 
+func (i *Infra) createOrUpdatePodDisruptionBudget(ctx context.Context, r ResourceRender) (err error) {
+	var (
+		pdb       *v1.PodDisruptionBudget
+		startTime = time.Now()
+		labels    = []metrics.LabelValue{
+			kindLabel.Value("PDB"),
+			nameLabel.Value(r.Name()),
+			namespaceLabel.Value(i.Namespace),
+		}
+	)
+
+	resourceApplyTotal.With(labels...).Increment()
+
+	if pdb, err = r.PodDisruptionBudget(); err != nil {
+		resourceApplyFailed.With(labels...).Increment()
+		return err
+	}
+
+	// when pdb is not set,
+	// then delete the object in the kube api server if got any.
+	if pdb == nil {
```

Comment:
shouldnt we just return here ?

Reply:
If a user removes a PDB section from an EnvoyProxy config, shouldn’t we remove the configured PDB?

Reply:
The deletePdb route is triggered only when a an EnvoyProxy (rendered resource) is removed. Therefore, it will not be activated in the event of proxy configuration changes.